### PR TITLE
ci: add workflow for yocto-check-layer

### PR DESCRIPTION
**Pull request by ricardosalveti**
Add PR-based workflow for calling yocto-check-layer on meta-qcom-hwe.

### CODE REVIEW

File: .github/workflows/check-layer.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Run yocto-check-layer
+        run: |
+          mkdir -p ../build
+          cd ../build
+          kas shell ../meta-qcom-hwe/ci/qcs6490-rb3gen2-core-kit.yml --command "yocto-check-layer-wrapper `pwd`/../meta-qcom-hwe --dependency `pwd`/poky/meta `pwd`/poky/meta-poky `pwd`/meta-qcom --no-auto-dependency"
```
Comment (quaresmajose) [inline .github/workflows/check-layer.yml:18]: The meta-poky is not a dependency of the BSP layer and is there mostly to provide the distro for CI testing

**Review by quaresmajose — COMMENTED** [review]
The yocto-check-layer don't need a fully conf directory populated and to avoid such case the yocto-check-layer-wrapper is used.
So we can run this job without kas as main propose of kas is configure the build, populating the conf directory.

**Comment by ricardosalveti** [comment]
> The yocto-check-layer don't need a fully conf directory populated and to avoid such case the yocto-check-layer-wrapper is used.
> So we can run this job without kas as main propose of kas is configure the build, populating the conf directory.

We need kas as it is responsible for fetching poky/oe-core and meta-qcom, both are needed dependencies.

**Comment by quaresmajose** [comment]
> > The yocto-check-layer don't need a fully conf directory populated and to avoid such case the yocto-check-layer-wrapper is used.
> > So we can run this job without kas as main propose of kas is configure the build, populating the conf directory.
> 
> We need kas as it is responsible for fetching poky/oe-core and meta-qcom, both are needed dependencies.

yeah, I forgot the the fetch step

**Review by quaresmajose — APPROVED** [review]
LGTM

**Pull request merged**